user-org-keycloak-util/src/main/java/org/sunbird/sso/service/impl/KeyCloakServiceImpl.java

```diff
@@ -0,0 +1,486 @@
+    UserRepresentation ur = null;
+    UserResource resource = null;
+    try {
+      resource = keycloak.realm(KeyCloakConnectionProvider.SSO_REALM).users().get(fedUserId);
+      ur = resource.toRepresentation();
+      if (ur == null) {
+        ProjectLogger.log(
+            "KeyCloakServiceImpl:updateUser: user not found with id " + fedUserId,
+            LoggerEnum.INFO.name());
+        return false;
+      }
+      mapUserTOKeyCloakUser(user, ur);
+      resource.update(ur);
+    } catch (Exception ex) {
+      ProjectLogger.log(
+          "KeyCloakServiceImpl:updateUser: update failure" + ex, LoggerEnum.ERROR.name());
+      throw new ProjectCommonException.ClientError(
+          IResponseMessage.INVALID_REQUESTED_DATA,
+          Localizer.getInstance().getMessage(IResponseMessage.INVALID_REQUESTED_DATA, null),
+          ResponseCode.CLIENT_ERROR.getCode());
+    }
+    return true;
+  }
+
+  /**
+   * Method to remove the user on basis of user id.
+   *
+   * @param request Map
+   * @return boolean true if success otherwise false .
+   */
+  @Override
+  public boolean removeUser(String userId) throws Exception {
+    Keycloak keycloak = KeyCloakConnectionProvider.getConnection();
+    try {
+      String fedUserId = getFederatedUserId(userId);
+      UserResource resource =
+          keycloak.realm(KeyCloakConnectionProvider.SSO_REALM).users().get(fedUserId);
+      if (resource != null) {
+        resource.remove();
+      } else {
+        ProjectLogger.log(
+            "KeyCloakServiceImpl:removeUser: user not found " + fedUserId, LoggerEnum.INFO.name());
+        return false;
+      }
+    } catch (Exception ex) {
+      ProjectLogger.log(
+          "KeyCloakServiceImpl:removeUser: remove user failure" + ex, LoggerEnum.ERROR.name());
+      throw new ProjectCommonException.ClientError(
+          IResponseMessage.INTERNAL_ERROR,
+          Localizer.getInstance().getMessage(IResponseMessage.INTERNAL_ERROR, null),
+          ResponseCode.SERVER_ERROR.getCode());
+    }
+    return true;
+  }
+
+  /**
+   * Method to deactivate the user on basis of user id.
+   *
+   * @param request Map
+   * @return boolean true if success otherwise false .
+   */
+  @Override
+  public boolean deactivateUser(String userId) {
+    return makeUserActiveOrInactive(userId, false);
+  }
+
+  /**
+   * Method to activate the user on basis of user id.
+   *
+   * @param request Map
+   * @return boolean true if success otherwise false .
+   */
+  @Override
+  public boolean activateUser(String userId) {
+    return makeUserActiveOrInactive(userId, true);
+  }
+
+  /**
+   * This method will take userid and boolean status to update user status
+   *
+   * @param userId String
+   * @param status boolean
+   * @throws ProjectCommonException
+   */
+  private boolean makeUserActiveOrInactive(String userId, boolean status) {
+    try {
+      String fedUserId = getFederatedUserId(userId);
+      validateUserId(fedUserId);
```

Comment:
this null check is not required , as this method is called internally so make sure whenever we are calling this method userid should not be null or empty

Reply:
it's not only caller responsibility to validate , method also need to validate.